internal/pkg/instance/logger.go

```diff
@@ -114,39 +111,68 @@ func (l *Logger) scanOutput(data []byte, atEOF bool) (advance int, token []byte,
 // NewWriter create a new pipe pair for corresponding stream.
 func (l *Logger) NewWriter(stream string, dropCRNL bool) *io.PipeWriter {
 	reader, writer := io.Pipe()
-	go l.scan(stream, reader, writer, dropCRNL)
+	closer := l.scan(stream, reader, writer, dropCRNL)
+	l.Lock()
+	l.closers = append(l.closers, closer)
+	l.Unlock()
 	return writer
 }
 
-func (l *Logger) scan(stream string, pr *io.PipeReader, pw *io.PipeWriter, dropCRNL bool) {
+func (l *Logger) scan(stream string, pr *io.PipeReader, pw *io.PipeWriter, dropCRNL bool) closer {
 	r := strings.NewReplacer("\r", "\\r", "\n", "\\n")
 	scanner := bufio.NewScanner(pr)
 	if !dropCRNL {
 		scanner.Split(l.scanOutput)
 	}
 
-	for scanner.Scan() {
-		l.fileMutex.Lock()
-		if !dropCRNL {
-			fmt.Fprint(l.file, l.formatter(stream, r.Replace(scanner.Text())))
-		} else {
-			fmt.Fprint(l.file, l.formatter(stream, scanner.Text()))
+	var wg sync.WaitGroup
```

Comment:
please see [the documentation for WaitGroup](https://godoc.org/sync#WaitGroup) and note where it says: "A WaitGroup must not be copied after first use."

By creating a closure on L#149, the code is copying the WaitGroup instance after the first use on L#128.

The proper fix for this is to use `wg := new(sync.WaitGroup)`

```suggestion
	wg := new(sync.WaitGroup)
```

Reply:
https://golang.org/ref/spec#Function_literals mentioned that :

> Function literals are closures: they may refer to variables defined in a surrounding function. Those variables are then shared between the surrounding function and the function literal, and they survive as long as they are accessible.

I may not fully understand what you mean by copy, because `scan` function can be called multiple time ?